### PR TITLE
[Fix] extraEnvs doesn't work for cluster operator helm charts 

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

_Select the type of your PR_

- Bugfix

### Description

This PR fixes the issue mentioned in #6145. It removes the use the use if `toYaml` which was giving error and makes use of `range`.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Could you please also add an example of the value to the README file or add it commented out to the values.yaml file?

This was of doing it also doesn't allow to reference secrets or confimaps etc. -> is that something we want? If we don't want it, does the format for passing the values still make sense? Would map be easier to use?

**Comment by ShubhamRwt** [comment]
@scholzj I tried to use `toYaml` with `nindent 12` and I was able to get through the error and get something like this 

```            - name: STRIMZI_FEATURE_GATES
              value: ""
            
            - name: JAVA_OPTS
              value: -Xms=256m -Xmx=256m
```

**Comment by ShubhamRwt** [comment]
I wonder why `indent` is not working fine while `nindent` is. It might be because
 ```           
            - name: STRIMZI_FEATURE_GATES
              value: {{ .Values.featureGates | quote }} ``` . We are not checking any `if` here and we are getting feature gates in our resources even when we dont use any

**Comment by ShubhamRwt** [comment]
I guess I got the fix

**Comment by ShubhamRwt** [comment]
Sorry looks like I made a mistake

**Comment by scholzj** [comment]
> We are not checking any `if` here and we are getting feature gates in our resources even when we dont use any

That alone is not wrong. We do not have to do if it makes things easier. But the empty value means the default behaviour => so it is not a problem on its own.

**Comment by ShubhamRwt** [comment]
Got it @scholzj . The changes I have pushed resolves the error which was coming.

**Review by spenczar — APPROVED** [review]
Duh, of course! Thanks for the fix.

`nindent` makes this a little cleaner to read, in my opinion, but it's no big deal.

**Comment by ShubhamRwt** [comment]
> Duh, of course! Thanks for the fix.
> 
> `nindent` makes this a little cleaner to read, in my opinion, but it's no big deal.

`nindent` will add an extra line too.

```- name: STRIMZI_FEATURE_GATES
              value: ""
            
            - name: JAVA_OPTS
              value: -Xms=256m -Xmx=256m
```

I guess we might not need it. What do you suggest @scholzj ?

**Comment by scholzj** [comment]
> I guess we might not need it. What do you suggest @scholzj ?

I do not know Helm, so I have no idea.

**Comment by scholzj** [comment]
@ppatierno Can you have a look at this if you have few minutes? Thanks.

File: packaging/helm-charts/helm3/strimzi-kafka-operator/templates/060-Deployment-strimzi-cluster-operator.yaml [hunk 1]
```diff
@@ -129,7 +129,7 @@ spec:
               value: {{ .Values.connectBuildTimeoutMs | quote }}
             {{- end }}
             {{- if .Values.extraEnvs }}
-            {{ toYaml .Values.extraEnvs | indent 12 }}
+{{ toYaml .Values.extraEnvs | indent 12 }}
```
Comment (ppatierno) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/templates/060-Deployment-strimzi-cluster-operator.yaml:132]: not an Helm expert but why the change is about changing the indentation of this line?

Comment (ShubhamRwt) [reply packaging/helm-charts/helm3/strimzi-kafka-operator/templates/060-Deployment-strimzi-cluster-operator.yaml:132]: It is due to the `indent 12`. I guess it starts counting the indentation space from the start

**Review by ppatierno — APPROVED** [review]

**Comment by rajha-korithrien** [comment]
Is not the proper Helm way to deal with this to use the "-" operator after the `{{`?

Notice that there is a difference between `{{` and `{{-`.

I believe that `{{-` instructs helm to remove all the leading whitespace of this line when rendering the template. Doing so will allow us to retain the 12 spaces while causing the results of the toYaml function to be correctly rendered into the resulting deployment yaml file.
